src/native/managed/cdacreader/src/Target.cs

```diff
@@ -352,6 +363,60 @@ public void ReadPointers(ulong address, Span<TargetPointer> buffer)
         }
     }
 
+    /// <summary>
+    /// Read a null-terminated UTF-8 string from the target
+    /// </summary>
+    /// <param name="address">Address to start reading from</param>
+    /// <returns>String read from the target</returns>}
+    public string ReadUtf8String(ulong address)
+    {
+        // Read characters until we find the null terminator
+        ulong end = address;
+        while (Read<byte>(end) != 0)
+        {
+            end += sizeof(byte);
+        }
+
+        int length = (int)(end - address);
+        if (length == 0)
+            return string.Empty;
+
+        Span<byte> span = stackalloc byte[length];
```

Comment:
I would avoid this approach. If there are issues here we could blow the stack and cause issues. I would instead perform `new byte[]`. I realize it unfortunate, but reliability is more important to me than perf in this section. Feel free to optimize this path when `length < 512` or so, but we need to check.